frontend/src/components/SideNavBarButton.tsx

```diff
@@ -1,14 +1,32 @@
+"use client";
 import React from 'react'
+import {useEffect, useRef} from 'react';
 
 interface SideNavBarButtonProps {
     svgIcon: React.ReactNode;
     text: string;
+    setCurrentSelection: (selection: string) => void;
+    currentText: string;
 }
-export default function SideNavBarButton({svgIcon, text}: SideNavBarButtonProps) {
+export default function SideNavBarButton({svgIcon, text, setCurrentSelection, currentText}: SideNavBarButtonProps) {
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  useEffect(() => {
```

Comment:
Do we really need useEffect here? I wouldn't think so. React should re-render as we update the state of the selection. This seems unnecessary.